client/landing/stepper/declarative-flow/internals/steps-repository/playground/lib/blueprint.ts

```diff
@@ -105,9 +105,17 @@ function getDefaultBlueprint( recommendedPhpVersion: string ): Blueprint {
 	};
 }
 
+export function getBlueprintName( name: string | null ): string | null {
+	if ( name && name in PREDEFINED_BLUEPRINTS ) {
+		return name;
+	}
+
+	return null;
+}
+
 async function getBlueprintFromUrl( recommendedPhpVersion: string ): Blueprint {
 	const url = new URL( window.location.href );
-	const predefinedBlueprintName = url.searchParams.get( 'blueprint' );
+	const predefinedBlueprintName = getBlueprintName( url.searchParams.get( 'blueprint' ) );
 
 	// If a predefined blueprint is specified and exists, use it
 	if ( predefinedBlueprintName && predefinedBlueprintName in PREDEFINED_BLUEPRINTS ) {
```

Comment:
This line can also be simplified now
```suggestion
	if ( predefinedBlueprintName ) {
```